libs/application-generic/src/usecases/get-subscriber-template-preference/get-subscriber-template-preference.usecase.ts

```diff
@@ -79,10 +86,11 @@ export class GetSubscriberTemplatePreference {
         templateId: command.template._id,
       });
 
-    const subscriberPreferenceChannels =
-      GetPreferences.mapWorkflowChannelPreferencesToChannelPreferences(
-        subscriberWorkflowPreferences,
-      ) || subscriberPreference?.channels;
+    const subscriberPreferenceChannels = subscriberWorkflowPreferences
+      ? GetPreferences.mapWorkflowChannelPreferencesToChannelPreferences(
+          subscriberWorkflowPreferences,
+        )
+      : subscriberPreference?.channels;
```

Comment:
Conditionally populating `subscriberPreferenceChannels` for both the V1 & V2 preferences, preferring the V2 preference if it exists.